### PR TITLE
Add support for Read, Update, and Delete for Messages

### PR DESCRIPTION
**Pull request by spang**
# Description
RUD support for Messages API

# JIRA Ticket
https://nylas.atlassian.net/browse/AV-2664

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by spang** [comment]
Unblocking my pet v3 side projects again! This could use a bit more testing—I've only verified that listing messages with `any_email` and `from` filters works so far. I'll continue to run it through its paces but I figured it's worth putting up a PR early so you know it exists. Figuring out how to properly deal with attributes that have the same names as Python keywords ("from", "in") was "fun"!

File: nylas/models/messages.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        "starred": NotRequired[bool],
+        "thread_id": NotRequired[str],
+        "received_before": NotRequired[int],
+        "received_after": NotRequired[int],
+        "has_attachment": NotRequired[bool],
+        "fields": NotRequired[Fields],
+        "search_query_native": NotRequired[str],
+    },
+)
+"""
```
Comment (spang) [inline nylas/models/messages.py:121]: FYI, I have no idea if this docstring will end up in the autogenerated docs. Couldn't figure out any way to include a docstring on a functionally declared TypedDict.

File: nylas/models/messages.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    folders: NotRequired[List[str]]
+
+
+class UpdateMessageRequest:
+
+    """Request payload for updating a message."""
+
+    unread: NotRequired[bool]
+    starred: NotRequired[bool]
+    folder: NotRequired[List[str]]
```
Comment (mrashed-dev) [inline nylas/models/messages.py:179]: I don't think there's any query/filter params for updating a message. For the `UpdateMessagRequest` the class hould also have `metadata` as a field that can be update and be a TypedDict:

```suggestion
class UpdateMessageRequest(TypedDict):

    """
    Request payload for updating a message.
    
    Attributes:
        starred: The message's starred status
        unread: The message's unread status
        folders: The message's folders
        metadata: A list of key-value pairs storing additional data
    """

    unread: NotRequired[bool]
    starred: NotRequired[bool]
    folder: NotRequired[List[str]]
    metadata: NotRequired[Dict[str, Any]]
```

File: nylas/resources/messages.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            response_type=Message,
+            query_params=query_params,
+        )
+
+    def update(
+        self,
+        identifier: str,
+        message_id: str,
+        request_body: UpdateMessageRequest,
+        query_params: UpdateMessageQueryParams,
```
Comment (mrashed-dev) [inline nylas/resources/messages.py:68]: Same as above, I don't think there's any query/filter params for updating a message

**Review by mrashed-dev — CHANGES_REQUESTED** [review]
Thanks for this PR @spang 💪🏼 Left some PR comments!

**Comment by spang** [comment]
Alright @mrashed-dev, I think this is ready to merge! I've tested all the basic calls and a subset of query params manually from a script—IMO full/comprehensive testing should be done by a dedicated test rig for the SDK or SDKs and operate off of the API spec.

Also, just pointing out that this PR doesn't include send support—so no C, just R-U-D! Send can follow in a separate PR.

**Review by mrashed-dev — APPROVED** [review]
@spang thank you so much for the PR & changes! Message Send + Drafts + Threads + Schedule Send + Smart Compose is currently being worked on, so soon we will have full messages support on the Python SDK 😄

**Pull request merged**